lib/src/license_detection/new_tokenizer.dart

```diff
@@ -0,0 +1,256 @@
+    //   return NewToken.fromSpan(_scanner.lastSpan!, tokenID++);
+    // }
+
+    // If none of the above conditions match, this implies
+    // the scanner is at single punctuation mark or leading
+    // punctuation in a word. Ignore them and move the scanner forward.
+    _scanner.readChar();
+    return null;
+  }
+
+  /// Scans through the input text and creates a list of [NewToken]
+  /// Whitespace, Leading or standalone punctuation are ignored as they are not significant.
+  /// But newLine token is stored to deal with list Items.\
+  while (!_scanner.isDone) {
+    final token = nextToken();
+
+    if (token != null) {
+      tokens.add(token);
+    }
+  }
+
+  /// Normalizes tokens using the approach of [google licenseClassifier]
+  /// to provide better chances of matching.
+  /// [google licenseClassifier]: https://github.com/google/licenseclassifier/blob/bb04aff29e72e636ba260ec61150c6e15f111d7e/v2/tokenizer.go#L34
+  tokens = _cleanNewTokens(tokens);
+
+  return tokens;
+}
+
+List<Token> _cleanNewTokens(List<Token> tokens) {
+  var output = <Token>[];
+  var tokenID = 0;
+  var firstInLine = true;
+
+  for (var token in tokens) {
+    // Ignore new line tokens for now.
+    // If accuracy of detection is low apply
+    // Guideline 2.1.4: Text that can be omited from license.
+    if (_newLineRegex.hasMatch(token.value)) {
+      firstInLine = true;
+      continue;
+    }
+
+    // Ignores list items.
+    if (firstInLine && _isListItem(token.value)) {
+      continue;
+    }
+
+    firstInLine = false;
+
+    final text = _cleanToken(token.value);
+
+    output.add(Token(text, tokenID++, token.line, token.fileSpan));
+  }
+
+  return output;
+}
+
+/// Normalizes the tokens using the approach of [google licenseClassifier]
+/// to provide better chances of matching.
+/// [google licenseClassifier]  https://github.com/google/licenseclassifier/blob/bb04aff29e72e636ba260ec61150c6e15f111d7e/v2/tokenizer.go#L85
+String _cleanToken(String tok) {
+  final runes = tok.runes;
+  var buffer = StringBuffer();
+
+  if (!_isLetter(runes.first)) {
+    if (_isDigit(runes.first)) {
+      runes.forEach((rune) {
+        if (_isDigit(rune) || rune == _dot || rune == _hiphen) {
+          buffer.write(String.fromCharCode(rune));
+        }
+      });
+
+      var text = buffer.toString();
+
+      if (text.runes.last == _dot) {
+        text = text.substring(0, text.length);
+      }
+
+      return text;
+    }
+  }
+
+  runes.forEach((rune) {
+    if (_isLetter(rune)) {
+      buffer.write(String.fromCharCode(rune));
+    }
+  });
+
+  var text = buffer.toString();
+
+  // If this is a varietal word normalize it according to
+  // Guideline 8.1.1: Legally equal words must be treated same.
+  text = _varietalWords[text] ?? text;
+  return text;
+}
+
+bool _isListItem(String token) {
+  final end = token[token.length - 1];
+  final start = token.substring(0, token.length - 1);
+
+  if ((end == '.' || end == ')') && _headers.contains(start)) {
```

Comment:
The `String` class has a lot of nice methods like [`String.endsWith`](https://api.dart.dev/stable/2.13.3/dart-core/String/endsWith.html)

Why not do:
```suggestion
  if ((token.endsWith('.') || token.endsWith(')')) && _headers.contains(start)) {
```